backends/ubpf/ubpfProgram.h

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 Orange
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef P4C_UBPFPROGRAM_H
+#define P4C_UBPFPROGRAM_H
+
+#include "target.h"
+#include "ubpfModel.h"
+#include "ir/ir.h"
+#include "frontends/p4/typeMap.h"
+#include "frontends/p4/evaluator/evaluator.h"
+#include "backends/ebpf/ebpfProgram.h"
+#include "codeGen.h"
+
+
+namespace UBPF {
+
+    class UBPFControl;
+    class UBPFParser;
+    class UBPFDeparser;
+    class UBPFTable;
+    class UBPFType;
+
+    class UBPFProgram : public EBPF::EBPFProgram {
+    public:
+        UBPFParser *parser{};
+        UBPFControl *control{};
+        UBPFDeparser *deparser{};
+        UBPFModel &model;
+
+        cstring headLengthVar, contextVar;
+
+        UBPFProgram(const CompilerOptions &options, const IR::P4Program *program,
+                    P4::ReferenceMap *refMap, P4::TypeMap *typeMap, const IR::ToplevelBlock *toplevel) :
+                EBPF::EBPFProgram(options, program, refMap, typeMap, toplevel), model(UBPFModel::instance) {
+            packetStartVar = cstring("pkt");
+            offsetVar = cstring("packetOffsetInBits");
+            headLengthVar = cstring("head_len");
+            contextVar = cstring("ctx");
+        }
+
+        bool build() override;
```

Comment:
can you remove some of these empty lines?

Reply:
I think it's done now.